### ISSUE TITLE
[1.0.0] `Series.filter` fails for for multidimensional keys in Spark mode

### ISSUE THREAD

**Issue by jwittenbach** [thunder-project/thunder#258]
The following works just fine:
```python
import thunder
series = thunder.series.fromrandom((100, 20), engine=sc)
# series.shape returns (100, 20)
series.filter(lambda v: v[0]>0).toarray()
```

However the following fails:
```python
import thunder
series = thunder.images.fromrandom((20, 10, 10), engine=sc).toseries()
#series.shape returns (10, 10, 20)
series.filter(lambda v: v[0]>0).toarray()
```

The problem stems from that fact that `Series.filter` implicitly assumes that only the first dimension is distributed. However, when coming from an `Images` object via `Images.toseries()`, the keys (and the distributed portion of the underlying `BoltArray`) remain multi-dimensional.

The fix:
Change to `Base._filter` and `Series.filter` so that  `Series.baseaxes` get passed along to `BoltArray.filter`

**Comment by jwittenbach** [comment]
After further consideration, a full fix to this problem actually relies on have a distributed sparse array -- something that `Bolt` does not currently support.

For now, one fix would be to add a `Series.flatten` method. This will linearize the keys so that filter can then subsequently be applied.

**Comment by jwittenbach** [comment]
Fixed with labels in #263
